utils/utils_daemon.go

```diff
@@ -3,14 +3,14 @@
 package utils
 
 import (
+	"github.com/docker/docker/pkg/system"
 	"os"
-	"syscall"
 )
 
 // IsFileOwner checks whether the current user is the owner of the given file.
 func IsFileOwner(f string) bool {
-	if fileInfo, err := os.Stat(f); err == nil && fileInfo != nil {
-		if stat, ok := fileInfo.Sys().(*syscall.Stat_t); ok && int(stat.Uid) == os.Getuid() {
+	if fileInfo, err := system.Lstat(f); err == nil && fileInfo != nil {
```

Comment:
We already had this discussion on a previous diff, and after giving it another thought, I'm afraid we took the wrong conclusions. I am able to give a different owner to a symlink compared to its target.

@gewoonrik I'm awfully sorry but I'm afraid we'd need to revert to system.Stat(), and rename the system.Stat struct to system.Stat_t like you suggested. Since I was the one who asked the change to Lstat initially, I have no problem carrying your PR now. Let me know, and sorry again :(